api/src/test/java/uk/gov/hmcts/payment/api/componenttests/PaymentGroupControllerTest.java

```diff
@@ -0,0 +1,126 @@
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.webAppContextSetup;
+
+@RunWith(SpringRunner.class)
+@ActiveProfiles({"local", "componenttest"})
+@SpringBootTest(webEnvironment = MOCK)
+public class PaymentGroupControllerTest {
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    private ServiceResolverBackdoor serviceRequestAuthorizer;
+
+    @Autowired
+    private UserResolverBackdoor userRequestAuthorizer;
+
+
+    private static final String USER_ID = UserResolverBackdoor.CITIZEN_ID;
+
+    private RestActions restActions;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    protected CustomResultMatcher body() {
+        return new CustomResultMatcher(objectMapper);
+    }
+
+    @Before
+    public void setup() {
+        MockMvc mvc = webAppContextSetup(webApplicationContext).apply(springSecurity()).build();
+        this.restActions = new RestActions(mvc, serviceRequestAuthorizer, userRequestAuthorizer, objectMapper);
+
+        restActions
+            .withAuthorizedService("divorce")
+            .withAuthorizedUser(USER_ID)
+            .withUserId(USER_ID)
+            .withReturnUrl("https://www.gooooogle.com");
+
+    }
+
+    @Test
+    @Transactional
+    public void testRetrievePaymentsByGroupReference() throws Exception {
+        CardPaymentRequest cardPaymentRequest = CardPaymentRequest.createCardPaymentRequestDtoWith()
+            .amount(new BigDecimal("100.1"))
+            .description("description")
+            .caseReference("telRefNumber")
+            .ccdCaseNumber("1234")
+            .service(Service.PROBATE)
+            .currency(CurrencyCode.GBP)
+            .provider("pci pal")
+            .channel("telephony")
+            .siteId("siteId")
+            .fees(Collections.singletonList(FeeDto.feeDtoWith()
+                .code("FEE0001")
+                .version("1")
+                .calculatedAmount(new BigDecimal("100.1"))
+                .build()))
+            .build();
+
+        MvcResult result1 = restActions
+            .withReturnUrl("https://www.google.com")
+            .withHeader("service-callback-url", "http://payments.com")
+            .post("/card-payments", cardPaymentRequest)
+            .andExpect(status().isCreated())
+            .andReturn();
+
+        PaymentDto paymentDto = objectMapper.readValue(result1.getResponse().getContentAsByteArray(), PaymentDto.class);
+        String reference = paymentDto.getReference();
+
+        // Retrieve payment by payment group reference
+        MvcResult result2 = restActions
+            .get("/payment-group/" + paymentDto.getPaymentGroupReference())
+            .andExpect(status().isOk())
+            .andReturn();
+
+        paymentDto = objectMapper.readValue(result2.getResponse().getContentAsByteArray(), PaymentDto.class);
+
+        assertThat(paymentDto).isNotNull();
```

Comment:
Optional: ideally it should be some thing like assertThat(paymentDto).isEqualToComparingOnlyGivenFields(expectedPaymentDto)